### PR TITLE
[bitnami/nats] adding pod disruption budget support

### PR DESCRIPTION
**Pull request by alexsn**
**Description of the change**
Adding pod disruption budget support to nats chart

**Benefits**
Provide ability to configure pdb for nats pods.

**Checklist**
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: bitnami/nats/templates/poddisruptionbudget.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    app: "{{ template "nats.name" . }}"
+    chart: "{{ template "nats.chart" . }}"
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+spec:
+  selector:
+    matchLabels:
+      app: "{{ template "nats.name" . }}"
+      release: {{ .Release.Name | quote }}
+{{ toYaml .Values.podDisruptionBudget | indent 2 }}
```
Comment (joancafom) [inline bitnami/nats/templates/poddisruptionbudget.yaml:16]: This works well for sure, but I think the most common approach followed by the majority of Charts is to split this into `minAvailable` and `maxUnavailable`. You can see an example in the [consul](https://github.com/bitnami/charts/blob/1f6bb260ebe2f43808939a60a307a24a124c1b24/bitnami/consul/templates/pod-dist-budget.yaml#L9) chart.

https://github.com/bitnami/charts/blob/1f6bb260ebe2f43808939a60a307a24a124c1b24/bitnami/consul/templates/pod-dist-budget.yaml#L9-L14

Could you please change this to follow an standardized approach?

File: bitnami/nats/templates/poddisruptionbudget.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{{- if .Values.podDisruptionBudget }}
```
Comment (joancafom) [inline bitnami/nats/templates/poddisruptionbudget.yaml:1]: Here as well, you may want to use the standard approach `.Values.pdb.create` (see my comment below)

**Comment by alexsn** [comment]
Thanks for the review, addressed all comments.

File: bitnami/nats/values-production.yaml [hunk 3]
```diff
@@ -72,6 +72,12 @@ podAnnotations: {}
 ##
 podLabels: {}
 
+## Pod disruption budget
+## ref: https://kubernetes.io/docs/tasks/run-application/configure-pdb/
+##
+pdb:
```
Comment (joancafom) [inline bitnami/nats/values-production.yaml:78]: You should also reflect the new changes here 😄

File: bitnami/nats/templates/poddisruptionbudget.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    app: "{{ template "nats.name" . }}"
+    chart: "{{ template "nats.chart" . }}"
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+spec:
+  selector:
+    matchLabels:
+      app: "{{ template "nats.name" . }}"
+      release: {{ .Release.Name | quote }}
+  {{- with .Values.pdb.minAvailable }}
```
Comment (joancafom) [inline bitnami/nats/templates/poddisruptionbudget.yaml:16]: While this is completely valid, could you please the following pattern (used in the rest of the charts with `pdb`):

```yml
  {{- if .Values.pdb.minAvailable }}
  minAvailable: {{ .Values.pdb.minAvailable }}
  {{- end }}
  {{- if .Values.pdb.maxUnavailable }}
  maxUnavailable: {{ .Values.pdb.maxUnavailable }}
  {{- end }}
```

These makes all Charts have an standardized approach and are easier to maintain 😁

Comment (alexsn) [reply bitnami/nats/templates/poddisruptionbudget.yaml]: Done even though I prefer to do less typing :)

**Pull request merged**